block/src/qcow/mod.rs

```diff
@@ -466,13 +476,17 @@ impl QcowFile {
         let direct_io = file.is_direct();
 
         let backing_file = if let Some(backing_file_path) = header.backing_file_path.as_ref() {
+            if max_nesting_depth == 0 {
+                return Err(Error::MaxNestingDepthExceeded);
+            }
             let path = backing_file_path.clone();
             let backing_raw_file = OpenOptions::new()
                 .read(true)
                 .open(path)
                 .map_err(Error::BackingFileIo)?;
-            let backing_file = crate::create_disk_file(backing_raw_file, direct_io)
-                .map_err(|e| Error::BackingFileOpen(Box::new(e)))?;
+            let backing_file =
+                crate::create_disk_file(backing_raw_file, direct_io, max_nesting_depth - 1)
```

Comment:
Why need to call `create_disk_file` vs `QCowFile::from_with_nesting_depth` ? QCOW backing files have to be other QCOW files. This means you can also avoid the change to `create_disk_file` completely

Reply:
My guess is that this was the way how crosvm's code was done, and it was borrowed as it was. I agree with Rob and believe this should be simplified. 

On top of this, there are several unused functions regarding to qcow backing file support and should be removed, as commented earlier (https://github.com/cloud-hypervisor/cloud-hypervisor/pull/6482#pullrequestreview-2089002509).

Reply:
I keep it because the backing can be non-qcow file. If we support only qcow as backing file, `QCowFile::from_with_nesting_depth` is better choice.